homeassistant/components/modbus/base_platform.py

```diff
@@ -90,6 +95,34 @@ def __init__(self, hub: ModbusHub, entry: dict[str, Any]) -> None:
         self._lazy_error_count = entry[CONF_LAZY_ERROR]
         self._lazy_errors = self._lazy_error_count
 
+        self._scan_group = entry.get(CONF_SCAN_GROUP)
+        self._unique_id = (
+            f"modbus_{hub.name}_{self._slave}_{self._input_type}_{self._address}"
```

Comment:
you cannot use hub.name as part of the unique_id as it is something the user can change.